openapi.yaml

```diff
@@ -16,175 +16,317 @@ host: api.oregonstate.edu
+        - name: q
+          in: query
+          type: string
           required: false
+          description: Query string to be used when searching through the contents of each note
+        - name: sortKey
+          in: query
           type: string
+          required: false
+          default: 'lastModified'
+          description: The key by which to sort the notes in the response
+        - name: sources
+          in: query
+          type: array
+          items:
+            type: string
+            enum: &SOURCES
+              - advisorPortal
+              - banner
+              - degreeWorks
+            collectionFormat: csv
+          required: false
+          description: A comma-separated list of sources to filter by
+        - name: contextTypes
+          in: query
+          type: array
+          items:
+            type: string
+            collectionFormat: csv
+          required: false
+          description: A comma-separated list of context types to filter by
       responses:
         200:
           description: Successful response
           schema:
-            $ref: '#/definitions/PetsResult'
+            $ref: '#/definitions/NotesResult'
         400:
-          description: Bad Request
+          description: Bad request
           schema:
             $ref: '#/definitions/Errors'
         500:
-          description: Internal Server Error
+          description: Internal server error
           schema:
             $ref: '#/definitions/Errors'
-  /pets/{id}:
-    get:
-      summary: Info for a specific pet
+    post:
+      summary: Create a new note
       tags:
-        - pets
-      description: Get pet by unique ID
-      operationId: getPetById
+        - notes
+      operationId: postNotes
       parameters:
-        - $ref: '#/parameters/authorization'
-        - name: id
-          in: path
-          description: Unique Pet ID
-          required: true
-          type: string
+        - in: body
+          name: requestBody
+          schema:
+            type: object
+            properties:
+              note:
+                $ref: '#/definitions/note'
+              studentID:
+                $ref: '#/definitions/studentID'
+              creatorID:
+                $ref: '#/definitions/creatorID'
+              permissions:
+                $ref: '#/definitions/permissions'
+              context:
+                $ref: '#/definitions/context'
+      responses:
+        200:
+          description: Successfully created a note
+          schema:
+            $ref: '#/definitions/NoteResult'
+        400:
+          description: Invalid note object
+          schema:
+            $ref: '#/definitions/Errors'
+        500:
+          description: Internal server error
+          schema:
+            $ref: '#/definitions/Errors'
+  /notes/{noteID}:
+    parameters:
+      - $ref: '#/parameters/noteID'
+    patch:
```

Comment:
Think about it more, I think it's a little strange to use `PATCH` method here. The behavior of `PATCH` should updating resource, but we are actually "adding" a new version `VersionHistoryResult`. I guess it will make more sense to use `POST /notes/{noteID}/versions` instead. We should probably discuss this topic as a group as well.